static/app/views/settings/organizationTeams/teamDetails.tsx

```diff
@@ -1,92 +1,48 @@
-import * as React from 'react';
+import {cloneElement, isValidElement, useState} from 'react';
 import {browserHistory, RouteComponentProps} from 'react-router';
 import styled from '@emotion/styled';
-import isEqual from 'lodash/isEqual';
 
 import {addErrorMessage, addSuccessMessage} from 'sentry/actionCreators/indicator';
-import {fetchTeamDetails, joinTeam} from 'sentry/actionCreators/teams';
+import {joinTeam} from 'sentry/actionCreators/teams';
 import {Client} from 'sentry/api';
 import Alert from 'sentry/components/alert';
 import Button from 'sentry/components/button';
 import IdBadge from 'sentry/components/idBadge';
 import ListLink from 'sentry/components/links/listLink';
-import LoadingError from 'sentry/components/loadingError';
 import LoadingIndicator from 'sentry/components/loadingIndicator';
 import NavTabs from 'sentry/components/navTabs';
 import SentryDocumentTitle from 'sentry/components/sentryDocumentTitle';
 import {t, tct} from 'sentry/locale';
 import TeamStore from 'sentry/stores/teamStore';
 import {Organization, Team} from 'sentry/types';
 import recreateRoute from 'sentry/utils/recreateRoute';
+import Teams from 'sentry/utils/teams';
 import withApi from 'sentry/utils/withApi';
 import withOrganization from 'sentry/utils/withOrganization';
-import withTeams from 'sentry/utils/withTeams';
 
 type Props = {
   api: Client;
-  teams: Team[];
   children: React.ReactNode;
   organization: Organization;
 } & RouteComponentProps<{orgId: string; teamId: string}, {}>;
 
-type State = {
-  loading: boolean;
-  error: boolean;
-  requesting: boolean;
-  team: Team | null;
-};
+function TeamDetails({api, children, ...props}: Props) {
+  const [currentTeam, setCurrentTeam] = useState(
+    TeamStore.getBySlug(props.params.teamId)
+  );
+  const [requesting, setRequesting] = useState<boolean>(false);
```

Comment:
```suggestion
  const [requesting, setRequesting] = useState(false);
```

don't need the type, it can be inferred from the parameter